### PR TITLE
Adds size cap for each individual log file, compresses on rollover

### PR DESCRIPTION
**Pull request by danhermann**
Addresses some of the requests in #7482:

- Verified locally that the size cap `SizeBasedTriggeringPolicy` functions properly with the `TimeBasedTriggeringPolicy` such that log rollovers occur when either the size cap or a time interval is reached.
- Sets a size cap of 100MB for each individual log file. This limit should be reviewed specifically for its suitability as a default.
- Gzips logs at the point of rollover.

It does not specify a limit on the number of log files to retain. That may be proposed in a separate PR since it would involve enabling the automatic deletion of log files by default in Logstash.

### CODE REVIEW

File: config/log4j2.properties [hunk 1]
```diff
@@ -15,25 +15,29 @@ appender.json_console.layout.eventEol = true
 appender.rolling.type = RollingFile
 appender.rolling.name = plain_rolling
 appender.rolling.fileName = ${sys:ls.logs}/logstash-${sys:ls.log.format}.log
-appender.rolling.filePattern = ${sys:ls.logs}/logstash-${sys:ls.log.format}-%d{yyyy-MM-dd}.log
+appender.rolling.filePattern = ${sys:ls.logs}/logstash-${sys:ls.log.format}-%d{yyyy-MM-dd}-%i.gz
```
Comment (yaauie) [inline config/log4j2.properties]: keeping the `log` as part of the suffix would be helpful; I'd recommend ending with `%d{yyyy-MM-dd}-%i.log.gz`.

Comment (danhermann) [reply config/log4j2.properties]: Done.

**Comment by danhermann** [comment]
@yaauie, I pushed a change to keep "log" as part of the extension of the compressed file.

**Comment by yaauie** [comment]
LGTM 🙌🏼

**Comment by jordansissel** [comment]
> It does not specify a limit on the number of log files to retain

I think the importance for users is around this idea: "don't run my machine out of disk space" -- and this PR doesn't do much for improving that.

If anything, I think this makes logging perhaps more challenging by creating even more files (many 100mb files per day, instead of what today's code of 1 per day).

I'm not against this PR, but I don't think it does much to alleviate the size problem?

**Pull request closed without merging**